File: Level3/Materal.BusinessFlow/Demo/BusinessFlowWebDemo/src/models/DataModelField/QueryDataModelFieldModel.ts
import { PageRequestModel } from "../PageRequestModel";
import { DataTypeEnum } from "./DataTypeEnum";

export class QueryDataModelFieldModel extends PageRequestModel {
    /**
     * 唯一标识
     */
    public ID?: string;
    /**
     * 名称
     */
    public Name?: string;
    /**
     * 数据模型ID
     */
    public DataModelID?: string;
    /**
     * 数据类型
     */
    public DataType?: DataTypeEnum;
    /**
     * 描述
     */
    public Description?: string;
}

File: Level3/Materal.BusinessFlow/Src/Materal.BusinessFlow.Abstractions/Services/Models/DataModelField/QueryDataModelFieldModel.cs
using Materal.Utils.Model;

namespace Materal.BusinessFlow.Abstractions.Services.Models.DataModelField
{
    public class QueryDataModelFieldModel : BaseQueryModel
    {
        /// <summary>
        /// 名称
        /// </summary>
        [Contains]
        public string? Name { get; set; }
        /// <summary>
        /// 数据模型唯一标识
        /// </summary>
        [Equal]
        public Guid? DataModelID { get; set; }
        /// <summary>
        /// 数据类型
        /// </summary>
        [Equal]
        public DataTypeEnum? DataType { get; set; }
        /// <summary>
        /// 描述
        /// </summary>
        [Contains]
        public string? Description { get; set; }
    }
}
